thunder/distributed/utils.py

```diff
@@ -84,8 +84,9 @@ def key(node: Node) -> int:
                 case wait_prim_impl.id | unpack_for_fsdp_prim_impl.id:
                     return len(order_in_trace)
                 case reduce_scatter_prim_impl.id | all_reduce_prim_impl.id:
-                    # Prefer larger communication ops over smaller ones
-                    return -node.bsym.args[0].numel
```

Comment:
That was a mistake on my part to add this condition. I didn't realize back then that this reordering might delay the start of computation. Thank you for fixing this!